### PR TITLE
Cut release version 0.41.0

### PR DESCRIPTION
**Pull request by s-urbaniak**
cc @coreos/team-monitoring

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,33 @@
...
+## 0.41.0 / 2020-07-29
+
+* [CHANGE] configmap-reload: Update to v0.4.0 (#3334)
+* [CHANGE] Update prometheus compatibility matrix to v2.19.2 (#3316)
+* [ENHANCEMENT] docs: update compatibility warning (#3369)
+* [ENHANCEMENT] Propagate Prometheus image version to statefulset (#3358)
+* [ENHANCEMENT] pkg: instrument resources being tracked by the operator (#3360)
+* [ENHANCEMENT] add SecretListWatchSelector to reduce memory and CPU footprint (#3355)
+* [ENHANCEMENT] Added support for configuring CA, cert, and key via secret or configmap. (#3249)
+* [ENHANCEMENT] go.mod: remove “replace” for client-go dep (#3332)
```
Comment (paulfantom) [inline CHANGELOG.md:12]: This is not user-facing. In the past, we didn't include such changes.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,33 @@
...
+* [ENHANCEMENT] consolidate image url logic (#3103)
```
Comment (paulfantom) [inline CHANGELOG.md:13]: We should be clearer here as this PR is essentially a deprecation notice for `tag`, `sha`, and `baseImage` parameters.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,33 @@
...
+* [ENHANCEMENT] pkg/namespace-labeler: fix whitespace (#3323)
```
Comment (paulfantom) [inline CHANGELOG.md:14]: Minor thing, definitely not user-facing.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,33 @@
...
+* [ENHANCEMENT] Support job name in probe (#3318)
+* [ENHANCEMENT] decouple alertmanager pod labels from selector labels (#3317)
+* [ENHANCEMENT] Fix probe ingress configs (#3312)
+* [ENHANCEMENT] pkg/prometheus: Ensure relabeling of container label in ServiceMonitors (#3315)
+* [ENHANCEMENT] Add E2E test for probe static targets (#3310)
+* [ENHANCEMENT] misc: remove v1beta1 crd remainings (#3311)
+* [ENHANCEMENT] Normalize default durations (#3308)
+* [ENHANCEMENT] pkg/prometheus: Allow enforcing namespace label in Probe configs (#3304)
+* [ENHANCEMENT] Add Synthetic Probes support (#2832)
```
Comment (paulfantom) [inline CHANGELOG.md:23]: This is a [FEATURE]

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,33 @@
...
+* [ENHANCEMENT] Support Prometheus vertical compaction (#3281)
```
Comment (paulfantom) [inline CHANGELOG.md:24]: I would say it's a FEATURE

File: CHANGELOG.md (hunk 1 above)
Comment (s-urbaniak) [reply CHANGELOG.md:12]: removed

File: CHANGELOG.md (hunk 3 above)
Comment (s-urbaniak) [reply CHANGELOG.md:14]: removed

File: CHANGELOG.md (hunk 4 above)
Comment (s-urbaniak) [reply CHANGELOG.md:23]: changed

File: CHANGELOG.md (hunk 5 above)
Comment (s-urbaniak) [reply CHANGELOG.md:24]: changed

File: CHANGELOG.md (hunk 2 above)
Comment (s-urbaniak) [reply CHANGELOG.md:13]: indeed, changed :+1:

**Comment by s-urbaniak** [comment]
@paulfantom addressed comments, ptal

**Comment by TechIsCool** [comment]
Probes are going to be great. Hoping this gets released today.

**Comment by brancz** [comment]
@TechIsCool we have 0 documentation on it so far, so if you could help us on that front that would be highly appreciated! :)

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,31 @@
 ## Next release
 
+## 0.41.0 / 2020-07-29
+
+* [CHANGE] configmap-reload: Update to v0.4.0 (#3334)
+* [CHANGE] Update prometheus compatibility matrix to v2.19.2 (#3316)
+* [FEATURE] Add Synthetic Probes support (#2832)
+* [FEATURE] Support Prometheus vertical compaction (#3281)
+* [ENHANCEMENT] docs: update compatibility warning (#3369)
```
Comment (brancz) [inline CHANGELOG.md:9]: we typically only list user facing changes affecting the release artifacts

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,31 @@
...
+* [ENHANCEMENT] Propagate Prometheus image version to statefulset (#3358)
+* [ENHANCEMENT] pkg: instrument resources being tracked by the operator (#3360)
+* [ENHANCEMENT] add SecretListWatchSelector to reduce memory and CPU footprint (#3355)
```
Comment (brancz) [inline CHANGELOG.md:12]: nit: let's start all of these with capital letters and make them full sentences.

Comment (s-urbaniak) [reply CHANGELOG.md:12]: Capitalized all entries. Prefixed changes (i.e. pkg: ...) have been capitalized at the beginning sentence.

File: CHANGELOG.md (hunk 6 above)
Comment (s-urbaniak) [reply CHANGELOG.md:9]: removed :+1:

**Comment by s-urbaniak** [comment]
@brancz thank you! addressed comments, ptal :)

**Comment by brancz** [comment]
lgtm

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,30 @@
...
+* [ENHANCEMENT] Consolidate image url logic, deprecating `baseImage`, `sha`, and `tag` in favor of `image` field in CRDs. (#3103)
+* [ENHANCEMENT] Support job name in probe (#3318)
+* [ENHANCEMENT] Decouple alertmanager pod labels from selector labels (#3317)
+* [ENHANCEMENT] Fix probe ingress configs (#3312)
+* [ENHANCEMENT] pkg/prometheus: Ensure relabeling of container label in ServiceMonitors (#3315)
+* [ENHANCEMENT] Add E2E test for probe static targets (#3310)
+* [ENHANCEMENT] misc: Remove v1beta1 crd remainings (#3311)
+* [ENHANCEMENT] Normalize default durations (#3308)
+* [ENHANCEMENT] pkg/prometheus: Allow enforcing namespace label in Probe configs (#3304)
+* [BUGFIX] test/framework: Fix prometheus active targets polling (#3361)
```
Comment (paulfantom) [inline CHANGELOG.md:22]: Not a user-facing bugfix

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,5 +1,30 @@
...
+* [ENHANCEMENT] Propagate Prometheus image version to statefulset (#3358)
+* [ENHANCEMENT] pkg: Instrument resources being tracked by the operator (#3360)
+* [ENHANCEMENT] Add SecretListWatchSelector to reduce memory and CPU footprint (#3355)
+* [ENHANCEMENT] Added support for configuring CA, cert, and key via secret or configmap. (#3249)
+* [ENHANCEMENT] Consolidate image url logic, deprecating `baseImage`, `sha`, and `tag` in favor of `image` field in CRDs. (#3103)
+* [ENHANCEMENT] Support job name in probe (#3318)
+* [ENHANCEMENT] Decouple alertmanager pod labels from selector labels (#3317)
+* [ENHANCEMENT] Fix probe ingress configs (#3312)
+* [ENHANCEMENT] pkg/prometheus: Ensure relabeling of container label in ServiceMonitors (#3315)
+* [ENHANCEMENT] Add E2E test for probe static targets (#3310)
```
Comment (paulfantom) [inline CHANGELOG.md:18]: tests are not user-facing

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,5 +1,30 @@
...
+* [CHANGE] Configmap-reload: Update to v0.4.0 (#3334)
+* [CHANGE] Update prometheus compatibility matrix to v2.19.2 (#3316)
+* [FEATURE] Add Synthetic Probes support (#2832)
+* [FEATURE] Support Prometheus vertical compaction (#3281)
+* [ENHANCEMENT] Propagate Prometheus image version to statefulset (#3358)
+* [ENHANCEMENT] pkg: Instrument resources being tracked by the operator (#3360)
+* [ENHANCEMENT] Add SecretListWatchSelector to reduce memory and CPU footprint (#3355)
+* [ENHANCEMENT] Added support for configuring CA, cert, and key via secret or configmap. (#3249)
+* [ENHANCEMENT] Consolidate image url logic, deprecating `baseImage`, `sha`, and `tag` in favor of `image` field in CRDs. (#3103)
+* [ENHANCEMENT] Support job name in probe (#3318)
```
Comment (paulfantom) [inline CHANGELOG.md:14]: We are adding probes support as a feature in this release. It seems weird to add enhancement to that feature in the same release. Maybe join this PR reference with FEATURE entry?

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,5 +1,30 @@
 ## Next release
 
+## 0.41.0 / 2020-07-29
+
+* [CHANGE] Configmap-reload: Update to v0.4.0 (#3334)
+* [CHANGE] Update prometheus compatibility matrix to v2.19.2 (#3316)
+* [FEATURE] Add Synthetic Probes support (#2832)
+* [FEATURE] Support Prometheus vertical compaction (#3281)
+* [ENHANCEMENT] Propagate Prometheus image version to statefulset (#3358)
```
Comment (paulfantom) [inline CHANGELOG.md:9]: nit: this is a part of consolidating image url logic.

File: CHANGELOG.md [hunk 12]
```diff
@@ -1,5 +1,30 @@
...
+* [ENHANCEMENT] pkg: Instrument resources being tracked by the operator (#3360)
+* [ENHANCEMENT] Add SecretListWatchSelector to reduce memory and CPU footprint (#3355)
+* [ENHANCEMENT] Added support for configuring CA, cert, and key via secret or configmap. (#3249)
+* [ENHANCEMENT] Consolidate image url logic, deprecating `baseImage`, `sha`, and `tag` in favor of `image` field in CRDs. (#3103)
+* [ENHANCEMENT] Support job name in probe (#3318)
+* [ENHANCEMENT] Decouple alertmanager pod labels from selector labels (#3317)
+* [ENHANCEMENT] Fix probe ingress configs (#3312)
```
Comment (paulfantom) [inline CHANGELOG.md:16]: Same as with `[ENHANCEMENT] Support job name in probe`. We are adding support for probes in the same release, so it isn't really an enhancement when looking from user perspective.

File: CHANGELOG.md [hunk 13]
```diff
@@ -1,5 +1,30 @@
...
+* [ENHANCEMENT] pkg/prometheus: Ensure relabeling of container label in ServiceMonitors (#3315)
+* [ENHANCEMENT] Add E2E test for probe static targets (#3310)
+* [ENHANCEMENT] misc: Remove v1beta1 crd remainings (#3311)
+* [ENHANCEMENT] Normalize default durations (#3308)
+* [ENHANCEMENT] pkg/prometheus: Allow enforcing namespace label in Probe configs (#3304)
+* [BUGFIX] test/framework: Fix prometheus active targets polling (#3361)
+* [BUGFIX] Revert "Normalize default durations" (#3364)
+* [BUGFIX] Reload alertmanager on configmap/secret change (#3319)
+* [BUGFIX] Fix probe promcfg (#3306)
```
Comment (paulfantom) [inline CHANGELOG.md:25]: Same as with `[ENHANCEMENT] Support job name in probe`.

File: CHANGELOG.md (hunk 8 above)
Comment (s-urbaniak) [reply CHANGELOG.md:22]: removed

File: CHANGELOG.md (hunk 9 above)
Comment (s-urbaniak) [reply CHANGELOG.md:18]: removed

File: CHANGELOG.md (hunk 10 above)
Comment (s-urbaniak) [reply CHANGELOG.md:14]: sounds good, unified :+1: and to me having an feature and a follow-up enhancement in the same release is fine.

File: CHANGELOG.md (hunk 11 above)
Comment (s-urbaniak) [reply CHANGELOG.md:9]: unified with 3103 enhancement (image logic consolidation)

**Comment by paulfantom** [comment]
Good point Lili! We do want info about move to new organization and deprecation of 
quay.io/coreos/prometheus-operator as well as quay.io/coreos/prometheus-config-reloader images in version v0.43 (2 releases from now). We should also announce new images in release v0.42. This should give enough time for users to migrate.

File: CHANGELOG.md (hunk 12 above)
Comment (s-urbaniak) [reply CHANGELOG.md:16]: unified with probes support :+1:

File: CHANGELOG.md (hunk 13 above)
Comment (s-urbaniak) [reply CHANGELOG.md:25]: unified with probes support :+1:

**Comment by s-urbaniak** [comment]
@paulfantom ptal, reflected all changes.

Proposal for the release notes:
```
Announcements:
1. prometheus-operator is going to be moved to the new `prometheus-operator` organization soon.
   The project will be hosted at https://github.com/prometheus-operator/prometheus-operator.
2. The `quay.io/coreos/prometheus-operator` image will be deprecedated in favor of
   `quay.io/prometheus-operator/promethus-operator` starting from 0.43.
2. The `quay.io/coreos/prometheus-config-reloader` image will be deprecedated in favor of
   `quay.io/prometheus-operator/prometheus-config-reloader` starting from 0.43.
```

I will put them as part of https://github.com/coreos/prometheus-operator/releases/tag/v0.41.0 once available.

**Comment by s-urbaniak** [comment]
One more pair of eyes before we merge? :-)

**Pull request merged**